Breakout/src/game.cpp

```diff
@@ -239,33 +243,15 @@ void Game::draw()
 
 void Game::run()
 {
-	sf::Clock clock;
-	sf::Time timer = clock.getElapsedTime();
-	sf::Time current = sf::Time::Zero;
-	sf::Time last = sf::Time::Zero;
-	float delta = 0.0f;
-	int frames = 0;
+	m_gameClock.initiateClock();
 
 	while (m_window.isOpen())
 	{
-		current = clock.getElapsedTime();
-		delta = (current - last).asSeconds();
-		last = current;
-
-		processInput(delta);
+		processInput(m_gameClock.elapsed());		
 		if (m_state != GameState::PAUSE)
 		{
-			update(delta);	
+			m_gameClock.update(); // gameclock.update() will call back all subscribed funcs when tick available
```

Comment:
I don't like the fact that a clock itself is responsible for updating the application. I would rather call `m_gameClock.update()` and `m_gameClock.readyToTick()` separately and call the game's update method here directly.

Reply:
The way you describe is actually how I've traditionally done it. What you say makes sense and I agree.